diff --git a/variants/xiao_s3_wio/XiaoS3WIOBoard.h b/variants/xiao_s3_wio/XiaoS3WIOBoard.h
index 7ae06a3..05e5589 100644
--- a/variants/xiao_s3_wio/XiaoS3WIOBoard.h
+++ b/variants/xiao_s3_wio/XiaoS3WIOBoard.h
@@ -10,4 +10,10 @@ public:
   const char* getManufacturerName() const override {
     return "Xiao S3 WIO";
   }
+
 };
+
+// LED polarity for Xiao S3 WIO
+#ifndef LED_STATE_ON
+#define LED_STATE_ON HIGH
+#endif
